sys/linux/security_ima.txt

```diff
@@ -0,0 +1,112 @@
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+#
+# It's better to disable syscall write$evm. An attempt to write to
+# "/sys/kernel/security/integrity/evm/evm" without additional vm setup
+# highly likely will result in "lost connection to test machine" error
+# due to -ENOKEY error.
+#
+# Recommended boot args for fuzzing: "ima_appraise=off evm=fix"
+
+
+include <uapi/linux/fcntl.h>
+include <security/integrity/evm/evm.h>
+
+openat$ima_measurements_ascii(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/security/integrity/ima/ascii_runtime_measurements"]], flags const[O_RDONLY], mode const[0]) fd
+openat$ima_measurements_binary(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/security/integrity/ima/binary_runtime_measurements"]], flags const[O_RDONLY], mode const[0]) fd
+
+resource fd_ima_count[fd]
+
+openat$ima_measurements_count(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/security/integrity/ima/runtime_measurements_count"]], flags const[O_RDONLY], mode const[0]) fd_ima_count
+openat$ima_violations(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/security/integrity/ima/violations"]], flags const[O_RDONLY], mode const[0]) fd_ima_count
+read$ima_count(fd fd_ima_count, buf ptr[out, fmt[dec, intptr]], count len[buf])
+
+ima_condition_elem [
+# base
+	func		fs_opt["func", stringnoz[ima_funcs]]
+	mask		fs_opt["mask", stringnoz[ima_masks]]
+	fsmagic		fs_opt_hex["fsmagic", intptr]
+	fsuuid		fs_opt["fsuuid", uuid_str]
+	uid_eq		fs_opt_dec["uid", uid]
+	euid_eq		fs_opt_dec["euid", uid]
+	fowner_eq	fs_opt_dec["fowner", uid]
+	uid_gt		fs_opt_nodelim["uid>", fmt[dec, uid]]
+	euid_gt		fs_opt_nodelim["euid>", fmt[dec, uid]]
+	fowner_gt	fs_opt_nodelim["fowner>", fmt[dec, uid]]
+	uid_lt		fs_opt_nodelim["uid<", fmt[dec, uid]]
+	euid_lt		fs_opt_nodelim["euid<", fmt[dec, uid]]
+	fowner_lt	fs_opt_nodelim["fowner<", fmt[dec, uid]]
+# lsm
+	obj_user	fs_opt_str["obj_user"]
+	obj_role	fs_opt_str["obj_role"]
+	obj_type	fs_opt_str["obj_type"]
+	subj_user	fs_opt_str["subj_user"]
+	subj_role	fs_opt_str["subj_role"]
+	subj_type	fs_opt_str["subj_type"]
+# option
+	appraise_type	fs_opt["appraise_type", stringnoz[ima_appraise_type]]
+	appraise_flag	stringnoz["appraise_flag=check_blacklist"]
+	permit_directio	stringnoz["permit_directio"]
+	keyrings	fs_opt["keyrings", ima_keyring_list]
+	pcr		fs_opt_dec["pcr", int64[0:64]]
+] [varlen]
+
+ima_condition {
+	cond	ima_condition_elem
+	sp0	const[' ', int8]
+} [packed]
+
+ima_policy_line {
+	action	stringnoz[ima_action]
+	sp0	const[' ', int8]
+	list	array[ima_condition]
+	n	const[0xa, int8]
+} [packed]
+
+ima_policy_list {
+	list	array[ima_policy_line]
+	z	const[0, int8]
+} [packed]
+
+ima_policy [
+	file	filename
+	policy	ima_policy_list
+] [varlen]
+
+resource fd_ima_policy[fd]
+openat$ima_policy(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/security/integrity/ima/policy"]], flags const[O_RDWR], mode const[0]) fd_ima_policy
```

Comment:
For some reason I get:
```
2020/07/24 20:15:11 disabling openat$ima_policy: open(/sys/kernel/security/integrity/ima/policy) failed: permission denied
```

However, this file is readable and writeable when I manually boot the same vm:
```
root@syzkaller:~# ls -la /sys/kernel/security/integrity/ima/policy
-rw------- 1 root root 0 Jul 24 17:39 /sys/kernel/security/integrity/ima/policy
echo dont_measure > /sys/kernel/security/integrity/ima/policy
[  103.093986][ T5729] ima: policy update completed
[  103.094230][   T28] audit: type=1807 audit(1595612451.977:2): action=dont_measure re1
[  103.098757][   T28] audit: type=1802 audit(1595612451.977:3): pid=5729 uid=0 auid=0 1
root@syzkaller:~# cat /sys/kernel/security/integrity/ima/policy
dont_measure
```

Reply:
> For some reason I get:
> 
> ```
> 2020/07/24 20:15:11 disabling openat$ima_policy: open(/sys/kernel/security/integrity/ima/policy) failed: permission denied
> ```
> 
> However, this file is readable and writeable when I manually boot the same vm:
> 
> ```
> root@syzkaller:~# ls -la /sys/kernel/security/integrity/ima/policy
> -rw------- 1 root root 0 Jul 24 17:39 /sys/kernel/security/integrity/ima/policy
> echo dont_measure > /sys/kernel/security/integrity/ima/policy
> [  103.093986][ T5729] ima: policy update completed
> [  103.094230][   T28] audit: type=1807 audit(1595612451.977:2): action=dont_measure re1
> [  103.098757][   T28] audit: type=1802 audit(1595612451.977:3): pid=5729 uid=0 auid=0 1
> root@syzkaller:~# cat /sys/kernel/security/integrity/ima/policy
> dont_measure
> ```

Weird.  You may add some logging to isSupportedOpenAt in pkg/host/syscalls_linux.go and then run with -debug flag.